src/System.Slices/src/System/Contract.cs

```diff
@@ -41,6 +41,18 @@ public static void RequiresInInclusiveRange(int start, int length)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void RequiresInInclusiveRange(int start, int length, int existingLength)
+        {
+            if ((uint)start > (uint)existingLength
+                || length < 0
+                || (uint)(start + length) > (uint)existingLength)
+            {
+                throw NewArgumentOutOfRangeException();
+            }
+        }
+
+        
```

Comment:
Maybe this API should take uint for length? This way the cast will be in the consumer side and so  the problem will be visible to the consumer.